File: Server/relaxo/src/main/java/de/teamrocket/relaxo/models/taskcomponent/TaskComponentFloat.java
package de.teamrocket.relaxo.models.taskcomponent;

import java.math.BigDecimal;

public class TaskComponentFloat extends TaskComponent {

    /**
     * Der Default-Wert.
     */
    private BigDecimal defaultValue;

    /**
     * Der minimale Wert.
     */
    private BigDecimal minValue;

    /**
     * Der maximale Wert.
     */
    private BigDecimal maxValue;

    /**
     * Die Nachkommastellen.
     */
    private int scale;

    public TaskComponentFloat() {
        this.setType("FLOAT");
    }

    /**
     * Gibt den Defaultwert zurück.
     *
     * @return Defaultwert der Komponente
     */
    public BigDecimal getDefaultValue() {
        return defaultValue;
    }

    /**
     * Setzt den Defaultwert der Komponente
     *
     * @param defaultValue Defaultwert der Komponente
     */
    public void setDefaultValue(BigDecimal defaultValue) {
        this.defaultValue = defaultValue;
    }

	public BigDecimal getMinValue() {
		return minValue;
	}

	public void setMinValue(BigDecimal minValue) {
		this.minValue = minValue;
	}

	public BigDecimal getMaxValue() {
		return maxValue;
	}

	public void setMaxValue(BigDecimal maxValue) {
		this.maxValue = maxValue;
	}

	public int getScale() {
		return scale;
	}

	public void setScale(int scale) {
		this.scale = scale;
	}

}

File: Client/Pikachu/PikachuLib/Models/TaskComponent/TaskComponentFloat.cs
using PikachuLib.Communication.Models.FormGroup;

namespace PikachuLib.Models.TaskComponent
{
    public class TaskComponentFloat : TaskComponent
    {
        /// <summary>
        /// Gibt den Wert zurück.
        /// </summary>
        public float Value { get; set; }

        /// <summary>
        /// Der Minimalwert der Zahl.
        /// </summary>
        public float MinValue { get; set; }

        /// <summary>
        /// Der Maximalwert der Zahl.
        /// </summary>
        public float MaxValue { get; set; }

        /// <summary>
        /// Die Anzahl der Nachkommastellen
        /// </summary>
        public int Scale { get; set; }

        public TaskComponentFloat(string name, string comment, bool required, int order)
            :base(name, comment, required, order)
        { }


        /// <summary>
        /// Überführt ein Kommunikationsmodel in das Model
        /// </summary>
        /// <param name="input">Das Model der Kommunikationsschicht.</param>
        /// <returns>Ein Flieskommazahl-Formularfeld.</returns>
        public static TaskComponentFloat Parse(FloatInputResponse input)
        {
            var comp = new TaskComponentFloat(input.Name, input.Comment, input.Required, input.Order)
            {
                Id = input.Id,
                Value = input.DefaultValue,
                MinValue = input.MinValue ?? 0,
                MaxValue = input.MaxValue ?? 0,
                ReadOnly =  input.Readonly,
                Scale = input.Scale
            };

            return comp;
        }
    }
}